crates/typst/src/math/attach.rs

```diff
@@ -469,3 +514,33 @@ fn compute_shifts_up_and_down(
 fn is_integral_char(c: char) -> bool {
     ('∫'..='∳').contains(&c) || ('⨋'..='⨜').contains(&c)
 }
+
+fn stretch_fragment(
+    ctx: &mut MathContext,
+    styles: StyleChain,
+    stretch: StretchInfo,
+    relative_to_width: Abs,
+    fragment: MathFragment,
+) -> MathFragment {
+    let glyph = match &fragment {
+        MathFragment::Glyph(glyph) => glyph.clone(),
+        MathFragment::Variant(variant) => {
+            GlyphFragment::new(ctx, styles, variant.c, variant.span)
```

Comment:
when it this the case? could we run into a case where we try to stretch both horizontally & vertically with a mix or `lr` and `stretch`?

Reply:
In a standalone `stretch(=, width: ...)`, the `MathFragment::Glyph` will be stretched and understandably, turned into a `MathFragment::Variant`. The same goes for `=^..._...`.

In `stretch(=, width: ...)^..._...` however, the construct of `stretch(=, width: ...)` will stretch the glyph and thus make it a `Variant`, and then the `^..._...` will also stretch it again. We can't combine this two stretching operations into one because of the order of parsing: `^..._...` is parsed _outside_ the node of `stretch(...)` call.